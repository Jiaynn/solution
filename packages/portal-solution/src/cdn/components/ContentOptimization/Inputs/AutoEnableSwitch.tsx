/**
 * @desc 自动启用 CDN 分发 开关
 * @author yaojingtian <yaojingtian@qiniu.com>
 */

import React from 'react'
import { observer } from 'mobx-react'

import Switch from 'react-icecream/lib/switch'
import Form from 'react-icecream/lib/form'

export type IValue = boolean

export interface IProps {
  value: IValue
  onChange(val: IValue): void
  formItemLayout?: IFormItemLayout
  simple?: boolean // 是否为简易模式（无 label & 无文字提示 (?)）
}

export const defaultFormItemLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 16 }
}

export default observer(function AutoEnableSwitchInput(props: IProps) {
  const switchRestProps = {
    checkedChildren: !props.simple ? '开启' : null,
    unCheckedChildren: !props.simple ? '关闭' : null
  }

  const switchComp = (
    <Switch
      checked={props.value}
      onChange={props.onChange}
      {...switchRestProps}
    />
  )

  if (props.simple) {
    return switchComp
  }

  const tips = (
    <ol>
      <li>1. 开启后，当视频文件瘦身成功时会自动启用 CDN 分发，将瘦身后的视频文件预取到各 CDN 节点；</li>
      <li>2. 通过本配置「已启用」CDN 分发的文件，可在「视频瘦身列表」的「操作」中单独停用。</li>
    </ol>
  )

  const formItemProps = {
    ...(props.formItemLayout || defaultFormItemLayout),
    label: '自动启用 CDN 分发',
    extra: tips
  }

  return (
    <Form.Item {...formItemProps}>
      {switchComp}
    </Form.Item>
  )
})
