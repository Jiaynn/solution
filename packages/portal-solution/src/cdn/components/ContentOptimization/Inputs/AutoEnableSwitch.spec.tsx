import React from 'react'

import { RendererUtils as Renderer } from 'test'

import AutoEnableSwitch from './AutoEnableSwitch'

it('renders correctly', () => {
  const renderer = new Renderer()

  const tree = renderer.createWithAct(
    <AutoEnableSwitch value onChange={() => null} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
