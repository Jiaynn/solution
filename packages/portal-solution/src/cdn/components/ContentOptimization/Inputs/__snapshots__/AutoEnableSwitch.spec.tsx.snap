// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-row ant-form-item"
>
  <div
    className="ant-col ant-col-6 ant-form-item-label"
  >
    <label
      className=""
      htmlFor={null}
      onClick={[Function]}
      title="自动启用 CDN 分发"
    >
      自动启用 CDN 分发
    </label>
  </div>
  <div
    className="ant-col ant-col-16 ant-form-item-control-wrapper"
  >
    <div
      className="ant-form-item-control"
    >
      <span
        className="ant-form-item-children"
      >
        <button
          aria-checked={true}
          checked={true}
          className="ant-switch ant-switch-checked"
          defaultChecked={false}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          role="switch"
          type="button"
        >
          <span
            className="ant-switch-inner"
          >
            开启
          </span>
        </button>
      </span>
      <div
        className="ant-form-extra"
      >
        <ol>
          <li>
            1. 开启后，当视频文件瘦身成功时会自动启用 CDN 分发，将瘦身后的视频文件预取到各 CDN 节点；
          </li>
          <li>
            2. 通过本配置「已启用」CDN 分发的文件，可在「视频瘦身列表」的「操作」中单独停用。
          </li>
        </ol>
      </div>
    </div>
  </div>
</div>
`;
